src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -3329,6 +3355,17 @@ private int indexOf(Rope sourceRope, Rope otherRope, int fromIndex) {
             }
             return -1;
         }
+
+        protected static boolean isSingleByteSearch(DynamicObject source, DynamicObject pattern) {
+            assert RubyGuards.isRubyString(source);
+            assert RubyGuards.isRubyString(pattern);
+
+            final Rope sourceRope = rope(source);
+            final Rope patternRope = rope(pattern);
+
+            return sourceRope.isSingleByteOptimizable() && RopeGuards.isSingleByteString(patternRope);
```

Comment:
That guard only checks the rope has 1 byte, but not that's it's singleByteOptimizable, is that always implied byteLength == 1?

Reply:
The string could be broken and have byteLength == 1. But in that case, the search would proceed the same way because the source string is single-byte optimizable.

Reply:
stringIndexBrokenCodeRange returns nil in all cases, that's not consistent witht he 1st specialization which could find an index for a pattern with broken CR.

Reply:
Good point. It's a bit tricky figuring out what the correct behavior is because it's primitive without any real specs. So you have to look at its usages. I'll update to maintain parity with the existing code.